### PR TITLE
Revert LocalGrpcListener usage of IDurableClient

### LINKED ISSUE #2581: String payloads are escaped differently between 1.0.2 and 1.0.3
### Description
We recently upgraded to Microsoft.Azure.Functions.Worker.Extensions.DurableTask version 1.0.3 a few days ago, and unfortunately, we encountered the same issue that was previously reported in issue #2577. Although a hotfix was applied to #2577, it led to unexpected behavior where the JSON data is now being read but with escaped characters like \\u0022. This is a departure from the behavior we observed in version 1.0.2.

An example of the JSON input before.
```
"{\"id\":\"73debd7d-e710-41cf-9388-9817c8efa517\"}"
````

### Expected behavior
The serialization of 1.0.3 behaves the same as 1.0.2 since I assume it shouldn't contain a breaking change. 

### Actual behavior
Instead of returning the payload 

```
"{\"id\":\"73debd7d-e710-41cf-9388-9817c8efa517\"}"
````

it returns the payload with an escaping

```
"{\\u0022id\\u0022:\\u002273debd7d-e710-41cf-9388-9817c8efa517\\u0022}"
````

### Relevant source code snippets
```csharp
// initiating the payload
var payload = "{\"id\":\"73debd7d-e710-41cf-9388-9817c8efa517\"}";
String instanceId = await client.ScheduleNewOrchestrationInstanceAsync("Orchestrator", payload);

// in the orchestrator reading the payload
public async Task<List<string>> RunOrchestrator([OrchestrationTrigger] TaskOrchestrationContext context)
{
        string? requestJson = context.GetInput<string>();
}
```

### Known workarounds
Downgrade to 1.0.2

### PR DESCRIPTION
**Pull request by jviau**
Reverts:
https://github.com/Azure/azure-functions-durable-extension/pull/2578
https://github.com/Azure/azure-functions-durable-extension/pull/2544

### Issue describing the changes in this PR

resolves #2581 

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)
* [x] My changes **do not** require any extra work to be leveraged by OutOfProc SDKs
    * [ ] Otherwise: That work is being tracked here: #issue_or_pr_in_each_sdk
* [ ] My changes **do not** change the version of the WebJobs.Extensions.DurableTask package
    * [x] Otherwise: major or minor version updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
* [x] My changes **do not** add EventIds to our EventSource logs
    * [ ] Otherwise: Ensure the EventIds are within the supported range in our existing Windows infrastructure. You may validate this with a deployed app's telemetry. You may also extend the range by completing a PR such as [this one](https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/7463263?_a=files).

### CODE REVIEW

File: test/SmokeTests/OOProcSmokeTests/DotNetIsolated/DotNetIsolated.csproj [hunk 1]
```diff
@@ -9,9 +9,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.18.0" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.19.0" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.13" />
-    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.13.0" OutputItemType="Analyzer" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.14.1" OutputItemType="Analyzer" />
     <PackageReference Include="Microsoft.DurableTask.Generators" Version="1.0.0-preview.1" OutputItemType="Analyzer" />
```
Comment (davidmrdavid) [inline test/SmokeTests/OOProcSmokeTests/DotNetIsolated/DotNetIsolated.csproj:15]: do these really need to be reversed?

**Review by davidmrdavid — APPROVED** [review]
approved w/ question

File: test/SmokeTests/OOProcSmokeTests/DotNetIsolated/DotNetIsolated.csproj (hunk 1 above)
Comment (jviau) [reply test/SmokeTests/OOProcSmokeTests/DotNetIsolated/DotNetIsolated.csproj:15]: Ah it isn't reverted - I actually upgraded them even _further_

Comment (davidmrdavid) [reply test/SmokeTests/OOProcSmokeTests/DotNetIsolated/DotNetIsolated.csproj:15]: Whoops, not enough coffee. Continue forth :)

**Pull request merged**